pkg/controller/attach_detach.go

```diff
@@ -496,8 +442,7 @@ func updateState(azVolumeAttachment *diskv1beta1.AzVolumeAttachment, state diskv
 		return nil, status.Errorf(codes.FailedPrecondition, "function `updateState` requires non-nil AzVolumeAttachment object.")
 	}
 	if mode == normalUpdate {
-		expectedStates := allowedTargetAttachmentStates[string(azVolumeAttachment.Status.State)]
-		if !containsString(string(state), expectedStates) {
+		if expectedStates, exists := allowedTargetAttachmentStates[string(azVolumeAttachment.Status.State)]; exists && !containsString(string(state), expectedStates) {
```

Comment:
If the current state is not in the map, we allow the state transition. Shouldn't the logic be `!exists || !containsString(...)`?

Reply:
That was the intention. Given that we cannot prefill our CRI with `OperationPendingState` on creation, we have to allow transition from an empty string to `Creating` or `Attaching` state. I thought about adding this relationship to the map but wasn't sure.

Reply:
Let's be explicit in the map. I worry that if we add a new state but forget to update the map, we may allow an arbitrary and invalid state transition.

Reply:
Will do 👍